nestkernel/connection.h

```diff
@@ -252,13 +252,19 @@ Connection< targetidentifierT >::check_connection_( Node& dummy_target,
   // this line might throw an exception
   source.send_test_event( dummy_target, receptor_type, get_syn_id(), true );
 
-  // 2. does the target accept the event type sent by sourc
+  // 2. does the target accept the event type sent by source
   // try to send event from source to target
   // this returns the port of the incoming connection
   // p must be stored in the base class connection
   // this line might throw an exception
   target_.set_rport( source.send_test_event( target, receptor_type, get_syn_id(), false ) );
 
+  // 3. do the events sent by source mean the same thing as they are
+  // interpreted in target?
+  if ( !( source.sends_signal() == all || target.receives_signal() == all )
+    && ( source.sends_signal() != target.receives_signal() ) )
+    throw IllegalConnection();
```

Comment:
Simplified the test to a simple bitwise and operation and negation. This should be a very minor performance penalty.